components/index_weaviate/tests/test_component.py

```diff
@@ -0,0 +1,63 @@
+import typing as t
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import weaviate
+from weaviate.embedded import EmbeddedOptions
+
+from src.main import IndexWeaviateComponent
+
+
+def get_written_objects(
+    client: weaviate.Client,
+    class_name: str,
+) -> t.List[t.Dict[str, t.Any]]:
+    """Taken from https://weaviate.io/developers/weaviate/manage-data/read-all-objects."""
+    query = (
+        client.query.get(
+            class_name,
+            ["id_", "passage"],
+        )
+        .with_additional(["id vector"])
+        .with_limit(8)
+    )
+
+    result = query.do()
+
+    return result["data"]["Get"][class_name]
+
+
+def test_index_weaviate_component(monkeypatch):
+    client = weaviate.Client(embedded_options=EmbeddedOptions())
+    monkeypatch.setattr("weaviate.Client", lambda *args, **kwargs: client)
```

Comment:
Why is this needed? I didn't need this in the [retrieval component test](https://github.com/ml6team/fondant/blob/main/components/retrieve_from_weaviate/tests/test_component.py).

Reply:
Missed the fact that we have access to a local URL, adjusted!